filter.go

```diff
@@ -0,0 +1,90 @@
+package inflight
+
+import (
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+)
+
+type FQFilter struct {
+	// list-watching API models
+	lock   *sync.Mutex
+	queues []*Queue
+
+	*sharedDispatcher
+
+	Matcher  func(*http.Request, []*Queue) *Queue
+	Delegate http.HandlerFunc
+}
+
+func (f *FQFilter) Serve(resp http.ResponseWriter, req *http.Request) {
+
+	matchedQueue := f.Matcher(req, f.queues)
+	// fmt.Printf("matchedQueue: %d\n", matchedQueue.Index)
+
+	dispatcher := f.sharedDispatcher.producers[matchedQueue.Priority]
+	func() {
+		dispatcher.lock.Lock()
+		defer dispatcher.lock.Unlock()
+		// dispatcher.queues[0].Priority
+
+		// fmt.Printf("dispatcher.requestsexecuting: %d\n", dispatcher.requestsexecuting)
+		if dispatcher.GetRequestsExecuting() > dispatcher.ACV {
+			// too many requests
+			// fmt.Println("throttled...")
+			resp.WriteHeader(http.StatusConflict)
+			return
+		}
+	}()
+	// TODO(aaron-prindle) currently the line below is a bit hacky..
+	distributionCh := dispatcher.fqScheduler.Enqueue(matchedQueue)
+	if distributionCh == nil {
+		// queues are full
+		// fmt.Println("throttled...")
+		resp.WriteHeader(http.StatusConflict)
+		return
+	}
+	select {
+	case finishFunc := <-distributionCh:
+		func() {
+			dispatcher.lock.Lock()
+			defer dispatcher.lock.Unlock()
+			dispatcher.requestsexecuting++
+		}()
+		defer finishFunc()
+
+		f.Delegate(resp, req)
+	// supports the timeout handlers context
+	case <-req.Context().Done():
+		resp.WriteHeader(http.StatusConflict)
```

Comment:
This is not a conflict, it is a timeout.
BTW, it probably does not matter what is written here, or whether anything _is_ written; the timeout filter is the one in charge of what really gets written in this case.